contracts/examples/mintable-erc1155-transfer/FxMintableERC1155RootTunnel.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {ERC1155} from "../../lib/ERC1155.sol";
+import {ERC1155Holder} from "../../lib/ERC1155Holder.sol";
+import {Create2} from "../../lib/Create2.sol";
+import {FxBaseRootTunnel} from "../../tunnel/FxBaseRootTunnel.sol";
+
+contract FxMintableERC1155RootTunnel is FxBaseRootTunnel, Create2, ERC1155Holder {
+    bytes32 public constant DEPOSIT = keccak256("DEPOSIT");
+    bytes32 public constant DEPOSIT_BATCH = keccak256("DEPOSIT_BATCH");
+    bytes32 public constant WITHDRAW = keccak256("WITHDRAW");
+    bytes32 public constant WITHDRAW_BATCH = keccak256("WITHDRAW_BATCH");
+    bytes32 public constant MAP_TOKEN = keccak256("MAP_TOKEN");
+
+    event TokenMappedERC1155(address indexed rootToken, address indexed childToken);
+    event FxWithdrawERC1155(
+        address indexed rootToken,
+        address indexed childToken,
+        address indexed userAddress,
+        uint256 id,
+        uint256 amount
+    );
+    event FxDepositERC1155(
+        address indexed rootToken,
+        address indexed depositor,
+        address indexed userAddress,
+        uint256 id,
+        uint256 amount
+    );
+    event FxWithdrawBatchERC1155(
+        address indexed rootToken,
+        address indexed childToken,
+        address indexed userAddress,
+        uint256[] ids,
+        uint256[] amounts
+    );
+    event FxDepositBatchERC1155(
+        address indexed rootToken,
+        address indexed userAddress,
+        uint256[] ids,
+        uint256[] amounts
+    );
+
+    mapping(address => address) public rootToChildTokens;
+    bytes32 public immutable childTokenTemplateCodeHash;
+
+    constructor(
+        address _checkpointManager,
+        address _fxRoot,
+        address _fxERC1155Token
+    ) FxBaseRootTunnel(_checkpointManager, _fxRoot) {
+        childTokenTemplateCodeHash = keccak256(minimalProxyCreationCode(_fxERC1155Token));
+    }
+
+    function mapToken(address rootToken) public {
+        require(rootToChildTokens[rootToken] == address(0x0), "FxMintableERC1155RootTunnel: ALREADY_MAPPED");
+
+        ERC1155 rootTokenContract = ERC1155(rootToken);
+        string memory uri = rootTokenContract.uri(0);
```

Comment:
Yeah, although I believe it can handled in various ways depending upon the token use case. Similar edge case with [ERC1155RootTunnel Example](https://github.com/fx-portal/contracts/blob/main/contracts/examples/erc1155-transfer/FxERC1155RootTunnel.sol#L60) ofcourse.

Reply:
I am not sure I understand what's the discussion about.